packages/neuron-wallet/src/utils/buffer.ts

```diff
@@ -22,4 +22,12 @@ export default class BufferUtils {
     const buf = Buffer.from(leHex.slice(2), 'hex')
     return (buf.readBigUInt64LE(8) << BigInt(64)) + buf.readBigUInt64LE(0)
   }
+
+  public static parseAmountFromSUDTData(leHex: string): bigint {
```

Comment:
It seems that hex string whose length is not 34 is invalid in the method and 0 will be returned. But according to the [RFC Draft](https://talk.nervos.org/t/rfc-simple-udt-draft-spec/4333)

>Simple UDT Rule 1: a SUDT cell must store SUDT amount in the first 16 bytes of cell data segment, the amount should be stored as little endian, 128-bit unsigned integer format. In the case of composable scripts, the SUDT amount must still be located at the initial 16 bytes in the data segment which corresponds to the composed SUDT script

The first 16 bytes will be treated as the amount field, namely the data whose length is greater than 34 should be regarded as valid.

Reply:
That is interesting to know. Great reference from the RFC. Any thoughts on this one @ququzone? Do you think we should treat it strictly 16 bytes or we can just parse the first 16 bytes and ignore the rest of the hex string?

Reply:
@Keith-CY @classicalliu 
Updated the code to support parsing for the first 16 bytes in the data field even when the field is greater than 16 bytes.